qiskit_machine_learning/neural_networks/opflow_qnn.py

```diff
@@ -118,18 +104,18 @@ def _construct_gradient_operator(self):
         except (ValueError, TypeError, OpflowError, QiskitError):
             logger.warning("Cannot compute gradient operator! Continuing without gradients!")
 
-    def _get_output_shape_from_op(self, op: OperatorBase) -> Tuple[int, ...]:
+    def _compute_output_shape(self, op: OperatorBase) -> Tuple[int, ...]:
         """Determines the output shape of a given operator."""
         # TODO: the whole method should eventually be moved to opflow and rewritten in a better way.
         # if the operator is a composed one, then we only need to look at the first element of it.
         if isinstance(op, ComposedOp):
-            return self._get_output_shape_from_op(op.oplist[0].primitive)
+            return self._compute_output_shape(op.oplist[0].primitive)
         # this "if" statement is on purpose, to prevent sub-classes.
         # pylint:disable=unidiomatic-typecheck
         if type(op) == ListOp:
             shapes = []
             for op_ in op.oplist:
-                shape_ = self._get_output_shape_from_op(op_)
+                shape_ = self._compute_output_shape(op_)
```

Comment:
```suggestion
                shapes = [self._compute_output_shape(op_) for op_ in op.oplist]
```

Reply:
Updated. Let's focus on the changes introduced in the PR.